pkg/config/config.go

```diff
@@ -169,6 +169,9 @@ func CacheOptions(logger logr.Logger, opts WatchOptions) cache.Options {
 						Name:   node.Name,
 						Labels: node.Labels,
 					},
+					Status: corev1.NodeStatus{
+						NodeInfo: node.Status.NodeInfo,
+					},
```

Comment:
Without this, calling `r.getNodeList` was returning nodes with an empty status so we couldn't check for Talos linux. This copies the node's NodeInfo into the cache so when we list them later, `NodeInfo` is populated.